litmus-portal/graphql-server/graph/schema.resolvers.go

```diff
@@ -54,7 +53,13 @@ func (r *mutationResolver) CreateChaosWorkFlow(ctx context.Context, input model.
 	return wfHandler.CreateChaosWorkflow(ctx, &input, data_store.Store)
 }
 
-func (r *mutationResolver) ReRunChaosWorkFlow(ctx context.Context, workflowID string) (string, error) {
+func (r *mutationResolver) ReRunChaosWorkFlow(ctx context.Context, projectID string, workflowID string) (string, error) {
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.ReRunChaosWorkFlow],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return "", err
+	}
```

Comment:
This change won't help secure the api because someone can send a different project id that is not linked with the workflow. The project id needs to be derived from the workflow being executed, what you need to do is retrieve the workflow data from the db, verify the workflow is valid and get the project id from there.

Reply:
cc: @SarthakJain26 @Saranya-jena

Reply:
Good catch! sorry for that!